vimrc.haskell

```diff
@@ -160,21 +73,26 @@ autocmd BufRead *
 " Haskell Lint
 nmap <silent> <leader>hl :Neomake hlint<CR>
 
-" Options for Haskell Syntax Check
-let g:neomake_haskell_ghc_mod_args = '-g-Wall'
-
-" }}}
+" " GHC errors and warnings
+nnoremap <silent> <leader>hc :call GhcideCompile()<cr>
+if !exists("g:ghcide_command")
+    let g:ghcide_command = "ghcide"
+endif
 
-" Point Conversion {{{
+function! GhcideCompile()
```

Comment:
I wonder whether it would be cleaner to use Vim's `:compiler` setting to specify that `:make` should call ghcide? You could configure `errorformat` so that Vim could parse ghcide's output to determine how to jump to errors/warnings in the original buffer.

See https://begriffs.com/posts/2019-07-19-history-use-vim.html#edit-compile-cycle

Reply:
Nice approach. I never used `:compiler` by my self, but I'll try it.

Reply:
What would you say to patch below?

```diff
diff --git a/vimrc.haskell b/vimrc.haskell
index f1eef24..a2c4717 100644
--- a/vimrc.haskell
+++ b/vimrc.haskell
@@ -73,26 +73,15 @@ autocmd BufRead *
 " Haskell Lint
 nmap <silent> <leader>hl :Neomake hlint<CR>
 
-" " GHC errors and warnings
-nnoremap <silent> <leader>hc :call GhcideCompile()<cr>
-if !exists("g:ghcide_command")
-    let g:ghcide_command = "ghcide"
+" GHC errors and warnings
+if exists("current_compiler")
+  finish
 endif
+let current_compiler = "ghcide"
 
-function! GhcideCompile()
-  " silent !clear
-  " execute "!" . g:ghcide_command . " " . bufname("%")
-  " Get the bytecode.
-  let bytecode = system(g:ghcide_command . " " . bufname("%") . " 2>&1")
-
-  " Open a new split and set it up.
-  vsplit __Ghcide_Bytecode__
-  normal! ggdG
-  setlocal filetype=potionbytecode
-  setlocal buftype=nofile
-
-  " Insert the bytecode.
-  call append(0, split(bytecode, '\v\n'))
-endfunction
+setlocal errorformat=%f:%l:%c:\ %t%*[a-zA-Z]:\ %m
+setlocal makeprg=ghcide\ %
 
+autocmd FileType * compiler ghcide
+nnoremap <silent> <leader>hc :make<cr>
 " }}}
```

* `,hc`-result:
![ghcide-compiler](https://user-images.githubusercontent.com/2326795/68581312-77c1d600-0478-11ea-86b7-02149c5a6028.png)